src/components/images/imageFetcher.js

```diff
@@ -1,21 +1,32 @@
 define(['dom'], function (dom) {
     'use strict';
 
+    // Firefox doesn't upload images to the GPU unless they are part of the DOM
+    // This hack eliminates the brief flicker while the image is uploaded
+
+    function createPreloader() {
+        var img = document.createElement('img');
+        img.style.position = 'fixed';
+        img.width = 5;
+        img.height = 5;
+        img.style.left = "-10px";
+        document.body.appendChild(img);
+        return img;
+    }
+
     function loadImage(elem, url) {
 
         if (!elem) {
             return Promise.reject('elem cannot be null');
         }
 
         if (elem.tagName !== "IMG") {
-
-            elem.style.backgroundImage = "url('" + url + "')";
-            return Promise.resolve();
-
-            //return loadImageIntoImg(document.createElement('img'), url).then(function () {
-            //    elem.style.backgroundImage = "url('" + url + "')";
-            //    return Promise.resolve();
-            //});
+            var preloader = createPreloader();
+            return loadImageIntoImg(preloader, url).then(function () {
+                document.body.removeChild(preloader);
+                elem.style.backgroundImage = "url('" + url + "')";
+                return Promise.resolve();
+            });
```

Comment:
@MrTimscampi did you remove image fetcher recently or was that a different module?

Reply:
#1065 removes the file, yes

Reply:
Would you rather try and remove the usage here before a merge or just fix it in #1065 when you work on that one again?

Reply:
It depends on what is merged first. It's pretty easy to fix in either case (Basically amounts to moving this code to ImageLoader)

I can take care of it when this is merged for sure.